[JSC] Implement `AsyncDisposableStack` prototype methods

sosukesuzuki opened the pull request:
#### 524caf808d1981a6f729d29622b7435383d1d3f9
<pre>
[JSC] Implement `AsyncDisposableStack` prototype methods
<a href="https://bugs.webkit.org/show_bug.cgi?id=293849">https://bugs.webkit.org/show_bug.cgi?id=293849</a>

Reviewed by Yusuke Suzuki.

This patch implements following `AsyncDisposableStack`&apos;s prototype methods
from Explicit Resource Management Proposal[1]:

- `adopt`[2]
- `defer`[3]
- `disposeAsync`[4]
- `disposed`[5]
- `move`[6]
- `use`[7]

[1]: <a href="https://github.com/tc39/proposal-explicit-resource-management">https://github.com/tc39/proposal-explicit-resource-management</a>
[2]: <a href="https://tc39.es/proposal-explicit-resource-management/#sec-asyncdisposablestack.prototype.adopt">https://tc39.es/proposal-explicit-resource-management/#sec-asyncdisposablestack.prototype.adopt</a>
[3]: <a href="https://tc39.es/proposal-explicit-resource-management/#sec-asyncdisposablestack.prototype.defer">https://tc39.es/proposal-explicit-resource-management/#sec-asyncdisposablestack.prototype.defer</a>
[4]: <a href="https://tc39.es/proposal-explicit-resource-management/#sec-asyncdisposablestack.prototype.disposeAsync">https://tc39.es/proposal-explicit-resource-management/#sec-asyncdisposablestack.prototype.disposeAsync</a>
[5]: <a href="https://tc39.es/proposal-explicit-resource-management/#sec-get-asyncdisposablestack.prototype.disposed">https://tc39.es/proposal-explicit-resource-management/#sec-get-asyncdisposablestack.prototype.disposed</a>
[6]: <a href="https://tc39.es/proposal-explicit-resource-management/#sec-asyncdisposablestack.prototype.move">https://tc39.es/proposal-explicit-resource-management/#sec-asyncdisposablestack.prototype.move</a>
[7]: <a href="https://tc39.es/proposal-explicit-resource-management/#sec-asyncdisposablestack.prototype.use">https://tc39.es/proposal-explicit-resource-management/#sec-asyncdisposablestack.prototype.use</a>

Canonical link: <a href="https://commits.webkit.org/295677@main">https://commits.webkit.org/295677@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/542c042261d0854fec133c21070e49e3e8c23f74

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/105778 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/25529 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/15922 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/110975 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/56374 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/26030 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/34032 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/80330 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/108784 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/20380 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/95453 "Passed tests") | [❌ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/60643 "Found 142 new API test failures: /WPE/TestUIClient:/webkit/WebKitWebView/query-permission-requests, /WPE/TestWebsiteData:/webkit/WebKitWebsiteData/deviceidhashsalt, /WPE/TestLoaderClient:/webkit/WebKitWebView/load-twice-and-reload, /WPE/TestCookieManager:/webkit/WebKitCookieManager/ephemeral, /WPE/TestWebKitWebView:/webkit/WebKitWebView/is-playing-audio, /WPE/TestConsoleMessage:/webkit/WebKitConsoleMessage/network-error, /WPE/TestWebsiteData:/webkit/WebKitWebsiteData/configuration, /WPE/TestCookieManager:/webkit/WebKitCookieManager/delete-cookies, /WPE/TestWebKitWebView:/webkit/WebKitWebView/terminate-unresponsive-web-process, /WPE/TestSSL:/webkit/WebKitWebView/insecure-content ... (failure)") | 
| | [💥 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/20083 "An unexpected error occured. Recent messages:OS: Sequoia (15.5), Xcode: 16.4; Running apply-patch; Checked out pull request; Skipped layout-tests; layout-tests (failure); Uploaded test results") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/13550 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/55813 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/98419 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/89814 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/13589 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/113824 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/104397 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/32918 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/24256 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/89409 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/33282 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/91684 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/89079 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/22720 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/33953 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/11758 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/28392 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/32843 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/38254 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/128709 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/32589 "Built successfully") | | [❌ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/35132 "Found 3 new JSC stress test failures: wasm.yaml/wasm/gc/ref-i31-eq.js.wasm-collect-continuously, wasm.yaml/wasm/gc/ref-i31-eq.js.wasm-eager-jettison, wasm.yaml/wasm/gc/ref-i31-eq.js.wasm-slow-memory (failure)") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/35938 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/34187 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

Constellation reviewed (commented):
Please add a entry to a file like this.

https://github.com/WebKit/WebKit/commit/c9ea94c29cbbb4af15681ea24012b2c5de40933f

sosukesuzuki committed:
[JSC] Implement `AsyncDisposableStack` prototype methods
https://bugs.webkit.org/show_bug.cgi?id=293849

Reviewed by Yusuke Suzuki.

This patch implements following `AsyncDisposableStack`'s prototype methods
from Explicit Resource Management Proposal[1]:

- `adopt`[2]
- `defer`[3]
- `disposeAsync`[4]
- `disposed`[5]
- `move`[6]
- `use`[7]

[1]: https://github.com/tc39/proposal-explicit-resource-management
[2]: https://tc39.es/proposal-explicit-resource-management/#sec-asyncdisposablestack.prototype.adopt
[3]: https://tc39.es/proposal-explicit-resource-management/#sec-asyncdisposablestack.prototype.defer
[4]: https://tc39.es/proposal-explicit-resource-management/#sec-asyncdisposablestack.prototype.disposeAsync
[5]: https://tc39.es/proposal-explicit-resource-management/#sec-get-asyncdisposablestack.prototype.disposed
[6]: https://tc39.es/proposal-explicit-resource-management/#sec-asyncdisposablestack.prototype.move
[7]: https://tc39.es/proposal-explicit-resource-management/#sec-asyncdisposablestack.prototype.use

Canonical link: https://commits.webkit.org/295677@main
